Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -66,15 +119,117 @@ class PreplannedMapModel: ObservableObject, Identifiable {
         }
     }
     
-    /// A Boolean value indicating if download can be called.
-    var canDownload: Bool {
-        switch status {
-        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
-                .downloading, .downloaded:
-            false
-        case .packaged, .downloadFailure:
-            true
+    /// Updates the status to downloaded if the mobile map pacakge exists.
+    @MainActor
+    private func updateDownloadStatus() {
+        self.mobileMapPackage = mapViewModel.mobileMapPackages.first(where: { $0.fileURL.lastPathComponent == preplannedMapArea.id?.rawValue.appending(".mmpk") })
+        
+        if mobileMapPackage != nil {
+            status = .downloaded
+        }
+    }
+    
+    /// Updates the status based on the download result of the mobile map package.
+    @MainActor
+    func updateDownloadStatus(for downloadResult: Optional<Result<MobileMapPackage, any Error>>) {
+        switch downloadResult {
+        case .success:
+            status = .downloaded
+        case .failure(let error):
+            status = .downloadFailure(error)
+        case .none:
+            updateDownloadStatus()
+        }
+    }
+    
+    /// Posts a local notification that the job completed.
+    func notifyJobCompleted(_ jobStatus: Job.Status) {
+        let content = UNMutableNotificationContent()
+        content.sound = UNNotificationSound.default
+        
+        if jobStatus == .succeeded {
+            content.title = "Download Succeeded"
+            content.body = "The job for City Hall Area has completed successfully."
+        } else if jobStatus == .failed {
+            content.title = "Download Failed"
+            content.body = "The job for City Hall Area failed."
+        }
+        
+        let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
+        let identifier = "My Local Notification"
+        let request = UNNotificationRequest(identifier: identifier, content: content, trigger: trigger)
+        
+        UNUserNotificationCenter.current().add(request)
+    }
+}
+
+extension PreplannedMapModel {
+    /// Downloads the given preplanned map area.
+    /// - Parameter preplannedMapArea: The preplanned map area to be downloaded.
+    /// - Precondition: `canDownload`
+    @MainActor
+    func downloadPreplannedMapArea() async {
```

Comment:
Ideally, this method can be further granulated

- Create directories and handle any file system errors
- Create download parameters and handle make parameter errors
- Create download task, run the task, and handle the result

This way, each separate logic can be easier unit tested.